src/figuresClasses.ts

```diff
@@ -1,19 +1,92 @@
-export interface Figure {
-
+export enum Shapes {
+    Triangle = 'triangle',
+    Circle = 'circle',
+    Rectangle = 'rectangle',
 }
 
-export class Triangle {
+export enum Colors {
+    Red = 'red',
+    Green = 'green',
+    Blue = 'blue', 
+}
 
+export interface Figure {
+    shape: Shapes;
+    color: Colors;
+    getArea(): number;
 }
 
-export class Circle {
 
+export class Triangle implements Figure {
+    shape = Shapes.Triangle;
+
+    constructor(
+        public color = Colors,
+        public a: number,
+        public b: number,
+        public c: number,
+    ) {
+        let sides: number[] = [a, b, c];
+
+        sides.sort((a, b) => b - a);
+
+        if (sides.some((side) => side <= 0)) {
+            throw Error('One side of the sides less then 0')
+        }
+
+        if (sides[0] >= sides[1] + sides[2]) {
+            throw Error('This is not Triangle');
+        }
+    }
+
+    getArea(): number {
```

Comment:
```suggestion
    getArea() {
```
you've already written it in the interface